src/adapters/ReactThirteenAdapter.js

```
@@ -0,0 +1,215 @@
+      EmptyElementType = instance._reactInternalInstance._renderedComponent._currentElement.type;
+    }
+    return EmptyElementType;
+  };
+})();
+
+const createShallowRenderer = function createRendererCompatible() {
+  const renderer = TestUtils.createRenderer();
+  renderer.render = (originalRender => function contextCompatibleRender(node, context = {}) {
+    ReactContext.current = context;
+    originalRender.call(this, React.createElement(node.type, node.props), context);
+    ReactContext.current = {};
+    return renderer.getRenderOutput();
+  })(renderer.render);
+  return renderer;
+};
+
+
+function instanceToTree(inst) {
+  if (typeof inst !== 'object') {
+    return inst;
+  }
+  const el = inst._currentElement;
+  if (!el) {
+    return null;
+  }
+  if (typeof el !== 'object') {
+    return el;
+  }
+  if (el.type === getEmptyElementType()) {
+    return null;
+  }
+  if (typeof el.type === 'string') {
+    const innerInst = inst._renderedComponent;
+    const children = innerInst._renderedChildren || { '.0': el._store.props.children };
+    return {
+      nodeType: 'host',
+      type: el.type,
+      props: el._store.props,
+      instance: inst._instance.getDOMNode(),
+      rendered: values(children).map(instanceToTree),
+    };
+  }
+  if (inst._renderedComponent) {
+    return {
+      nodeType: 'class',
+      type: el.type,
+      props: el._store.props,
+      instance: inst._instance || inst._hostNode || null,
+      rendered: instanceToTree(inst._renderedComponent),
+    };
+  }
+  throw new Error('Enzyme Internal Error: unknown instance encountered');
+}
+
+class SimpleWrapper extends React.Component {
+  render() {
+    return this.props.node || null;
+  }
+}
+
+SimpleWrapper.propTypes = { node: PropTypes.node.isRequired };
+
+class ReactThirteenAdapter extends EnzymeAdapter {
+  createMountRenderer(options) {
+    const domNode = options.attachTo || global.document.createElement('div');
```

if `createMountRenderer` in a non-dom env is going to throw, can we have an explicit throw instead of accessing a property off of undefined?